### PR TITLE
Enable JupyterHandlers to set the logger on log_request

### LINKED ISSUE #273: Display proper classname in logs
The logs emitted from extensions are appearing as ServerApp logs. Instead, these logs should reflect the proper LoggingConfigurable class name that emitted those logs. i.e. LabApp, NotebookApp, etc.

### PR DESCRIPTION
**Pull request by Zsailer**
This enables server extensions to log from their own logger, rather than always logging from the `ServerApp`'s logger. Previously, all log messages from extensions were routed through the ServerApp logger, leading to some confusion on where these messages were actually coming from.

Fixes #273.

### CODE REVIEW

File: jupyter_server/serverapp.py [hunk 1]
```diff
@@ -273,7 +273,7 @@ def init_settings(self, jupyter_app, kernel_manager, contents_manager,
             server_root_dir=root_dir,
             jinja2_env=env,
             terminals_available=False,  # Set later if terminals are available
-            serverapp=self
+            serverapp=jupyter_app
```
Comment (echarles) [inline jupyter_server/serverapp.py:276]: Is this related to the changes for the logger? Just curious trying to understand the logic.

Comment (Zsailer) [reply jupyter_server/serverapp.py:276]: Yeah. This was actually a bug that was exposed in PR (line 42). The serverapp was not actually being passed to the extension handlers; rather, the tornado web application was being passed unintentionally.

**Comment by Zsailer** [comment]
Thanks, @echarles and @blink1073 for the review!
